### ISSUE TITLE
Better diagnostic on where error is occurred

### ISSUE THREAD

**Issue by AlexanderZobkov** [asciidoctor/asciidoctor-pdf#446]
During convertion from asciidoc to pdf, various page layouting errors could occur. For example, a table can't fit page:
[asciidoctor-prawn-cant-fit.txt](https://github.com/asciidoctor/asciidoctor-pdf/files/268979/asciidoctor-prawn-cant-fit.txt)

```
[user@vm]$ ../bin/asciidoctor-pdf -d book --trace doc.adoc                                     
/home/user/.rvm/gems/ruby-2.2.1/gems/prawn-2.1.0/lib/prawn/text/formatted/line_wrap.rb:199:in `update_output_based_on_last_fragment': Prawn::Errors::CannotFit (Prawn::Errors::CannotFit)
        from /home/user/.rvm/gems/ruby-2.2.1/gems/prawn-2.1.0/lib/prawn/text/formatted/line_wrap.rb:190:in `fragment_finished'                                                           
        from /home/user/.rvm/gems/ruby-2.2.1/gems/prawn-2.1.0/lib/prawn/text/formatted/line_wrap.rb:112:in `block in add_fragment_to_line' 
...
        from /home/user/.rvm/gems/ruby-2.2.1/gems/asciidoctor-1.5.4/lib/asciidoctor/cli/invoker.rb:86:in `each'
        from /home/user/.rvm/gems/ruby-2.2.1/gems/asciidoctor-1.5.4/lib/asciidoctor/cli/invoker.rb:86:in `invoke!'
        from ../bin/asciidoctor-pdf:31:in `<main>'
```

Currently it's impossible to determine problematic lines of the input asciidoc file just by looking on a stacktrace.
The issue can be experienced by a person who:
- converts existing asciidoc files to PDF in addition to HTML
- migrates from MS Word with involvement of semi-automatic conversion tools like `pandoc`

In the both use-cases, there could be many problematic regions of asciidoc markups in existing asciidoc files that can't be correctly layouted in PDF for this or that reason. And it will be hard to find these regions to adjust the problematic markup.

**Comment by mojavelinux** [comment]
In many cases, we have the line and file information, so we can put that into the log message. However, right now we don't look for and catch a lot of exceptions, so they just bubble up to the top level. We'll have to study the cases individually to understand better where to put these catch blocks.

In general, table rendering in the PDF converter is very rudimentary, and pandoc is known to create some pretty horrendous AsciiDoc. The combination of those two things means the most of your errors are likely going to be around tables until we address #6. It's just a really tough problem to solve.

**Comment by AlexanderZobkov** [comment]
The issue (probably related to #6 or similar issue) that I can demonstrate is the following:
1. a table that has several columns (6)
2. one cell includes inline anchors

Attempts to render such table result in `Prawn::Errors::CannotFit` error. 
However after:
--  removing anchors
-- or reducing number of columns in the table but keep anchors
the table can be succesfully rendered. In both cases anchors are not rendered as text in the table.

I suspect that Prawn or the converted includes anchors when determining whether this or that element can fit a page or a table.

Should I create a new issue for that?

**Comment by mojavelinux** [comment]
Please. File an issue and we'll get to the bottom of it.

**Comment by mojavelinux** [comment]
In Asciidoctor PDF 2, the cannot fit error has been changed to a warning that includes diagnostic information (if sourcemap is enabled) in the fix for #2009. Inline images are also resized to fit, which was addressed by #1798. Therefore, I'm marking this as a duplicated covered by those other scenarios. If problems still remain, please file a new issue and describe the scenario. Each issue should be focused on a single problem.
